plugin/core/sessions.py

```diff
@@ -897,6 +904,20 @@ def run_code_action_async(self, code_action: Mapping[str, Any]) -> Promise:
         # At this point it cannot be a command anymore, it has to be a proper code action.
         # A code action can have an edit and/or command. Note that it can have *both*. In case both are present, we
         # must apply the edits before running the command.
+        if self.has_capability("codeActionProvider.resolveSupport"):
+            # TODO: Should we accept a SessionBuffer? What if this capability is registered with a documentSelector?
+            # We must first resolve the command and edit properties, because they can potentially be absent.
+            promise = self.request_promise_async(Request("codeAction/resolve", code_action))
```

Comment:
I'm thinking that maybe this resolve call could be avoided if all properties of the item from the `codeActionProvider.resolveSupport` list are already resolved or missing. But I'm not sure it's worth the extra complexity. I guess we can wait and see how servers use it.

Reply:
> I guess we can wait and see how servers use it.

Agreed.